### ISSUE TITLE
Support form ARM Neon

### ISSUE THREAD

**Issue by AudriusButkevicius** [minio/blake2b-simd#15]
First of all, this is great, thank you very much for producing this.

Secondly, could ARM platforms benefit from an implementation which uses Neon?

**Comment by harshavardhana** [comment]
> First of all, this is great, thank you very much for producing this.
> 
> Secondly, could ARM platforms benefit from an implementation which uses Neon?

ARM would work by using Native Go implementation won't have any SIMD benefits yet.

**Comment by AudriusButkevicius** [comment]
Yeah, but Neon would bring it in par with performance of other platforms, and potentially beat whatever other hashes are available in the stdlib.

Anyways, this has the right tags, so now lets hope for a wild neon enthusiast to appear :)

**Comment by abperiasamy** [comment]
Looks like Raspberry Pi 3 has NEON support :)

**Comment by fwessels** [comment]
Here are some pointers for NEON stuff:
- [BLAKE2_NEON_Compress32](https://github.com/weidai11/cryptopp/blob/master/blake2.cpp#L3465)
- [BLAKE2_NEON_Compress64](https://github.com/weidai11/cryptopp/blob/master/blake2.cpp#L3971)
- [Rust implementation of BLAKE2 with SIMD optimizations](https://github.com/cesarb/blake2-rfc#simd-optimization)
